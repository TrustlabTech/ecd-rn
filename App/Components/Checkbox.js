/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import React, { Component } from 'react'
import {
  View,
  Text,
  TouchableHighlight,
  Switch
} from 'react-native'
import { Colours, FontSizes } from '../GlobalStyles'
/**
 * A configurable Checkbox
 * @example
 * <Checkbox
 *  style={'switch'}
 *  key={i}
 *  width={300}
 *  text={"Example text"}
 *  onPress={ () => someAction() }
 *  checked={true}
 * />
 */
export default class Checkbox extends Component {

  render () {
    let checkbox = null
    switch (this.props.style) {

      case 'classic':
        checkbox =
          <TouchableHighlight
            onPress={() => this.props.onPress(!this.props.checked)}
            style={{
              borderRadius: 5
            }}
          >
            <View style={{
              alignItems: 'center',
              justifyContent: 'center',
              backgroundColor: 'white',
              borderColor: Colours.primaryLowlight,
              borderWidth: 1,
              borderStyle: 'solid',
              borderRadius: 5,
              padding: 1,
              width: this.props.height || 45,
              height: 45 }}
            >
              <Text>{this.props.checked ? 'X' : ''}</Text>
            </View>
          </TouchableHighlight>
        break

      case 'switch':
        checkbox =
          <Switch
            value={this.props.checked}
            ref={(ref) => { this._switch = ref }}
            onValueChange={() => this.props.onPress()}
          />
        break

      default:
        checkbox =
          <Switch
            value={this.props.checked}
            ref={(ref) => { this._switch = ref }}
            onValueChange={() => this.props.onPress()}
          />
        break
    }

    return (
      <View style={{
        flex: 1,
        marginLeft: 10,
        marginRight: 10,
        marginTop: 5,
        marginBottom: 5,
        height: 45}}
      >
        <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}>

          <View style={{ flex: 1 }}>

            <Text style={{ color: Colours.primary, fontSize: FontSizes.h5, textAlign: 'left', marginLeft: 10 }}>
              {this.props.text}
            </Text>
          </View>
          {checkbox}

        </View>
      </View>
    )
  }
}

Checkbox.propTypes = {
  height: React.PropTypes.number,
  text: React.PropTypes.string,
  checked: React.PropTypes.bool,
  onPress: React.PropTypes.func,
  style: React.PropTypes.object
}
