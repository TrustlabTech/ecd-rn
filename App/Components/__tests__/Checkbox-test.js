import 'react-native'
import React from 'react'
import Checkbox from '../Checkbox'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <Checkbox text="Test Checkbox"/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})