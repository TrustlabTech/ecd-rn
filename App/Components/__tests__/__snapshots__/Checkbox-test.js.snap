exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 45,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#393939",
            "fontSize": 22,
            "marginLeft": 10,
            "textAlign": "left",
          }
        }>
        Test Checkbox
      </Text>
    </View>
    <RCTSwitch
      disabled={false}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          undefined,
        ]
      }
      value={false} />
  </View>
</View>
`;
